mi/dataset/driver/zplsc_c/zplsc_echogram_generator.py

```diff
@@ -0,0 +1,633 @@
+                        month = int(date_dir[1])
+
+                        # Save the date and indicate that the entire month should be generated.
+                        echogram_dates[date(year, month, 1)] = True
+
+        return echogram_dates, date_dirs_url
+
+    def get_deployment_dirs(self, subsite):
+        """
+        This method will determine the deployment directories for the subsite
+        passed in.
+
+        :param subsite: The subsite of the ZPLSC instrument.
+        :return: deployments: The list of deployment directories.
+        """
+
+        deployments = [deployment for deployment in self.deployments]
+        if not deployments:
+            # Generate the portion of the URL up to the subsite directory.
+            subsite_url = join(RAW_DATA_URL, subsite.upper())
+            try:
+                # Get the all the deployment sub-directories under the subsite directory.
+                deployment_dirs = self.get_dir_contents(subsite_url)
+            except urllib2.HTTPError:
+                raise
+
+            # Generate the list of the 24 1-hour raw data files for the given date.
+            deployment_list = DEPLOYMENT_DIR_MATCHER.findall(deployment_dirs)
+            deployment_list.sort()
+
+            if self.process_mode:
+                deployment_list = [deployment_list[-1]]
+
+            for deployment in deployment_list:
+                try:
+                    deployments.append(int(deployment))
+
+                except ValueError as ex:
+                    log.error('Invalid deployment number: %s: %s', deployment, ex.message)
+                    break
+
+        return deployments
+
+    def store_echogram(self, zplsc_echogram_file):
+        """
+        This method will store the generated echogram in the location specified
+        in the configuration file. It will then clean up the temporary folder.
+
+        :param zplsc_echogram_file: The generated ZPLSC echogram
+        :return:
+        """
+
+        # Create the echogram directory, then copy the echogram to the local directory.
+        self.echogram_directory = expanduser(self.echogram_directory)
+        if not exists(self.echogram_directory):
+            mkdir(self.echogram_directory)
+
+        system('cp ' + zplsc_echogram_file + ' ' + self.echogram_directory)
+
+        # Clean up the temporary files.
+        system('rm -f ' + join(self.temp_directory, '*'))
+
+    def generate_zplsc_echograms(self):
+        """
+        This method will get the subsites, deployments and dates from the
+        command line or all of the subsites, deployments and dates for the
+        daily process.  It will generate the echograms based on those inputs
+        and upload the echograms to the raw data server.
+
+        :return:
+        """
+
+        # Create the temporary data file directory.
+        self.temp_directory = join(expanduser(USER_HOME), TEMP_DIR)
+        if not exists(self.temp_directory):
```

Comment:
Check to see if there is already a python module for temporary directory/file management. I think it will simplify this and also make it more robust.

Reply:
There is, it's tempfile.py.  I looked into it and it does have some nice features, but I don't think it would be beneficial here.  I would only be able to use the temporary directory.  The temporary file feature is useless here because the files in temporary directory are the raw data files, the concatenated data file.  The temporary directory is created in an obscure file path under /var.  I have an option on the tool that allows you to save the raw data and concatenated data files.  So, I'd like for the temporary directory be in the users home directory, so it is more user friendly if someone wants to examine the raw data files.